src/formatting/reorder.rs

```diff
@@ -226,8 +226,12 @@ fn rewrite_reorderable_or_regroupable_items(
             for (item, list_item) in normalized_items.iter_mut().zip(list_items) {
                 item.list_item = Some(list_item.clone());
             }
-            if context.config.merge_imports() {
-                normalized_items = merge_use_trees(normalized_items);
+            match context.config.imports_merge_style() {
+                ImportMergeStyle::Crate => normalized_items = merge_use_trees(normalized_items),
+                ImportMergeStyle::Module => {
+                    normalized_items = unnest_use_trees(merge_use_trees(normalized_items))
```

Comment:
Is there a way we could format with the Module merge style without first converting it to the Crate style? I realize that could potentially involve some non-minimal refactoring of the existing code to split out the reusable bits from the Crate-specific strategy, but I think we'll want to go that route given the additional strategies we know we'll need to support eventually

Reply:
Alternatively if you'd prefer, I'd be okay with just adding `// TODO` comments inline here and deferring this to a later date. The new config option and mapping of the old ones is a good enhancement in its own right, so don't necessarily need to block

Reply:
Yeah, I'm not a huge fan of this code either. I'll try to rewrite it.